data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,301 @@
+        MetadataInserted(AssetId),
+        MetadataRemoved(AssetId),
+    }
+
+    impl Identifiable for AssetEvent {
+        type Id = AssetId;
+    }
+
+    impl IdTrait for AssetEvent {
+        fn id(&self) -> &AssetId {
+            match self {
+                Self::Created(id)
+                | Self::Deleted(id)
+                | Self::Increased(id)
+                | Self::Decreased(id)
+                | Self::MetadataInserted(id)
+                | Self::MetadataRemoved(id) => id,
+            }
+        }
+    }
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum AssetDefinitionEvent {
+        Created(AssetDefinitionId),
+        Deleted(AssetDefinitionId),
+        MetadataInserted(AssetDefinitionId),
+        MetadataRemoved(AssetDefinitionId),
+    }
+
+    impl Identifiable for AssetDefinitionEvent {
+        type Id = AssetDefinitionId;
+    }
+
+    impl IdTrait for AssetDefinitionEvent {
+        fn id(&self) -> &AssetDefinitionId {
+            match self {
+                Self::Created(id)
+                | Self::Deleted(id)
+                | Self::MetadataInserted(id)
+                | Self::MetadataRemoved(id) => id,
+            }
+        }
+    }
+}
+
+mod peer {
+    //! This module contains `PeerEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum PeerEvent {
+        Trusted(PeerId),
+        Untrusted(PeerId),
+    }
+
+    impl Identifiable for PeerEvent {
+        type Id = PeerId;
+    }
+
+    impl IdTrait for PeerEvent {
+        fn id(&self) -> &PeerId {
+            match self {
+                Self::Trusted(id) | Self::Untrusted(id) => id,
+            }
+        }
+    }
+}
+
+#[cfg(feature = "roles")]
+mod role {
+    //! This module contains `RoleEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum RoleEvent {
+        Created(RoleId),
+        Deleted(RoleId),
+    }
+
+    impl Identifiable for RoleEvent {
+        type Id = RoleId;
+    }
+
+    impl IdTrait for RoleEvent {
+        fn id(&self) -> &RoleId {
+            match self {
+                Self::Created(id) | Self::Deleted(id) => id,
+            }
+        }
+    }
+}
+
+mod account {
+    //! This module contains `AccountEvent` and its impls
+
+    use super::*;
+
+    /// Account event
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum AccountEvent {
+        Asset(AssetEvent),
+        Created(AccountId),
+        Deleted(AccountId),
+        Authentication(AccountId),
+        Permission(AccountId),
```

Comment:
I think event variants should be **verbs**. Many different permission events are packed under this variant but I think that I would at least differentiate between `PermissionAdded`/`PermissionRemoved`

Reply:
maybe for authentication as well